src/inference/include/openvino/runtime/intel_gna/properties.hpp

```diff
@@ -264,12 +264,47 @@ inline std::istream& operator>>(std::istream& is, PWLDesignAlgorithm& pwl_design
 static constexpr Property<PWLDesignAlgorithm> pwl_design_algorithm{"GNA_PWL_DESIGN_ALGORITHM"};
 
 /**
- * @brief The option to allow to specify the maximum error percent that the optimized algorithm finding
- * will be used to find PWL functions.
- * By default (in case of NO value set), 1.0 value is used.
+ * @brief Enum to define pwl approximation mode.
  * @ingroup ov_runtime_gna_prop_cpp_api
  */
-static constexpr Property<float> pwl_max_error_percent{"GNA_PWL_MAX_ERROR_PERCENT"};
+enum class PWLApproximationMode {
+    PERFORMANCE = 1,  //!<  Optimize for max performance, may apply properties which slightly affect accuracy
+    ACCURACY = 2,     //!<  Optimize for max accuracy
```

Comment:
cosmetic: do we need '= 2' here?

Reply:
Theoretically not, but what is the problem with giving exact value?
Especially that it aligns with convention of other declarations in the file.

Reply:
having initial value only, enum's new values do not require to maintain them - less error prone. If C++ provides feature of autoincrement of enum values, why not use it.
By the way, explicit values are usable for bitfields for example.
I'm not insisting on though. please choose whatever you prefer.